web_generator/lib/src/interop_gen/sub_type.dart

```diff
@@ -227,63 +318,35 @@ Type getSubTypeOfTypes(List<Type> types,
     return t;
   }
 
-  // get the type hierarchy of each type.
+  // Calculate the intersection of all type hierarchies
   final typeMaps = types.map(getTypeHierarchy);
   final parentHierarchy = typeMaps.map((map) => map.expand());
   final commonTypes =
       parentHierarchy.reduce((val, element) => val.intersection(element));
 
-  // filter any
-  final markedForRemoval = <String>{};
-  for (final type in commonTypes) {
-    // map should be same for all types
-    final lookupResult = typeMaps.first.lookup(type)!;
-    final typeMap = typeMaps.first.getMapWithLookup(lookupResult.path);
-
-    // remove types in map
-    markedForRemoval.addAll(commonTypes.where(
-        (t) => typeMap.lookup(t) != null && typeMap.lookup(t)!.level != 0));
+  final topoList = topologicalList(typeMaps.toList());
+  for (final level in topoList) {
+    final typesAtLevel = commonTypes.intersection(level);
+    if (typesAtLevel.isNotEmpty) {
```

Comment:
I'd just briefly leave a comment here mentioning that by definition, another least common ancestor can't be in the next level.